custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,411 @@
+
+class JimakuSubtitle(Subtitle):
+    '''Jimaku Subtitle.'''
+    provider_name = 'jimaku'
+    
+    hash_verifiable = False
+
+    def __init__(self, language, video, download_url, filename):
+        super(JimakuSubtitle, self).__init__(language, page_link=download_url)
+        
+        self.video = video
+        self.download_url = download_url
+        self.filename = filename
+        self.release_info = filename
+        self.is_archive = filename.endswith(accepted_archive_formats)
+        
+    @property
+    def id(self):
+        return self.download_url
+
+    def get_matches(self, video):
+        matches = set()
+        
+        # Episode/Movie specific matches
+        if isinstance(video, Episode):
+            if sanitize(video.series) and sanitize(self.video.series) in (
+                    sanitize(name) for name in [video.series] + video.alternative_series):
+                matches.add('series')
+            
+            if video.season and self.video.season is None or video.season and video.season == self.video.season:
+                matches.add('season')
+        elif isinstance(video, Movie):
+            if sanitize(video.title) and sanitize(self.video.title) in (
+                    sanitize(name) for name in [video.title] + video.alternative_titles):
+                matches.add('title')
+
+        # General matches
+        if video.year and video.year == self.video.year:
+            matches.add('year')
+
+        video_type = 'movie' if isinstance(video, Movie) else 'episode'
+        matches.add(video_type)
+        
+        guess = guessit(self.filename, {'type': video_type})
+        logger.debug(f"Guessit: {guess}")
+        for g in guess:
+            if g[0] == "release_group" or "source":
+                if video.release_group == g[1]:
+                    matches.add('release_group')
+                    break
+                
+        # Prioritize .srt by repurposing the audio_codec match
+        if self.filename.endswith(".srt"):
+            matches.add('audio_codec')
+
+        return matches
+
+class JimakuProvider(Provider):
+    '''Jimaku Provider.'''
+    video_types = (Episode, Movie)
+    
+    api_url = 'https://jimaku.cc/api'
+    api_ratelimit_max_delay_seconds = 5
+    api_ratelimit_backoff_limit = 3
+    
+    corrupted_file_size_threshold = 500
+    
+    languages = {Language.fromietf("ja")}
+
+    def __init__(self, enable_name_search_fallback, enable_archive_fallback, enable_ai_subs, api_key):
+        if api_key:
+            self.api_key = api_key
+        else:
+            raise ConfigurationError('Missing api_key.')
+
+        self.enable_name_search_fallback = enable_name_search_fallback
+        self.download_archives = enable_archive_fallback
+        self.enable_ai_subs = enable_ai_subs
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
```

Comment:
Why we are not using the SZ_USER_AGENT?

Reply:
Honestly I just copied that off some other provider when I started, but if `SZ_USER_AGENT` is preferred then I'll just use that instead.

Reply:
Now using `SZ_USER_AGENT` instead.